### PR TITLE
WIP add cleanup function to test database setup.

### PR DESCRIPTION
**Pull request by trusch**

### CODE REVIEW

**Comment by trusch** [comment]
damn, its a go1.14 feature

**Comment by pragmader** [comment]
I just was going to ask you how did you test it.

**Comment by trusch** [comment]
@pragmader it compiled just fine for me, I didn't expect that test cleanup is a bleeding edge feature :D

**Comment by pragmader** [comment]
@trusch I meant to test how it mitigates the issues with a lot of open connections in Hub. You need my branch and to decrease max connection count.

**Comment by trusch** [comment]
@pragmader I did not test that, but the current code obviously is missing the connection closing so I just added that :man_shrugging:

I can build a test directly here in go base, that shouldn't be that hard.

**Comment by pragmader** [comment]
This code makes sense only if it mitigates the issue. Otherwise, we should not care about closing because it's running in tests anyway. AFAIK the tests are running in separate processes per package, so not closing connection in unit tests of a package should be a problem. But it seems like in Hub there are more than 100 calls of this function now and if this change mitigates the issue it's cool, otherwise it does not make sense to remove support for go < 1.14.

File: pkg/db/test/db.go [hunk 1]
```diff
@@ -74,6 +75,13 @@ func GetDatabase(t *testing.T, inits ...DBInitializer) (name string, testDB *sql
 			return "", nil
 		}
 	}
+	t.Cleanup(func() {
```
Comment (LucasRoesler) [inline pkg/db/test/db.go]: I didn't know about Cleanup

.... Also, looking at the rest of the docs, should we start marking these utility functions with `Helper()`

https://golang.org/pkg/testing/#T.Helper

> Helper marks the calling function as a test helper function. When printing file and line information, that function will be skipped. Helper may be called simultaneously from multiple goroutines.

**Comment by trusch** [comment]
> does not make sense to remove support for go < 1.14.

Thats totally convincing, but I wouldn't say that we simply shouldn't care about closing because its "only" in the tests. If we can make the test setup code more robust, that would be a good thing, especially since we are using this code in nearly all our projects. If we would have a way of closing correctly you would never have run into this issue and spend time figuring out what was going on. 

For now its ok to just don't care because the alternatives would be 1) dropping support for anything < go1.14 or 2) breaking the API by introducing a context argument which would also require code changes in all our projects.

**Comment by LucasRoesler** [comment]
I think we have lived long enough with out, should we close it?

**Pull request closed without merging**